appinventor/appengine/src/com/google/appinventor/client/DesignToolbar.java

```diff
@@ -94,16 +94,19 @@ public DesignProject(String name, long projectId) {
 
     // Returns true if we added the screen (it didn't previously exist), false otherwise.
     public boolean addScreen(String name, FileEditor formEditor, FileEditor blocksEditor) {
-      if (!screens.containsKey(name)) {
-        screens.put(name, new Screen(name, formEditor, blocksEditor));
-        return true;
-      } else {
+      if (screens.containsKey(name)) {
         return false;
       }
+      screens.put(name, new Screen(name, formEditor, blocksEditor));
+      ((YaBlocksEditor) blocksEditor).addScreen(name);
```

Comment:
Rather than doing this, it might be better to have `YaBlocksEditor` implement `ProjectChangeListener` and have it register with with `Project` object held by its corresponding `YaProjectEditor`. This way each editor can receive updates when a node is added or removed without tightly coupling the behavior here.